be/src/exec/hash_join_node.cpp

```diff
@@ -518,6 +541,18 @@ pipeline::OpFactories HashJoinNode::decompose_to_pipeline(pipeline::PipelineBuil
     return lhs_operators;
 }
 
+pipeline::OpFactories HashJoinNode::decompose_to_pipeline(pipeline::PipelineBuilderContext* context) {
+    using namespace pipeline;
+    if (runtime_state()->enable_spill() &&
+        (_join_type == TJoinOp::INNER_JOIN || _join_type == TJoinOp::LEFT_SEMI_JOIN)) {
```

Comment:
In current phase, only suport INNER_JOIN, LEFT_SEMI_JOIN?  do you have a plan to support other join types in future? please add TODO to hint this. 
It is very common in CTAS, ITAS that plan contains LEFT/FULL/RIGHT OUTER JOIN types, which is a drastic pain.

Reply:
@stdpain add a  TODO hint